autoconsent/autoconsent-store/src/main/java/com/duckduckgo/autoconsent/store/AutoconsentSettingsDataStore.kt

```diff
@@ -27,17 +27,25 @@ interface AutoconsentSettingsDataStore {
 
 class RealAutoconsentSettingsDataStore constructor(
     private val context: Context,
-    private val onByDefault: Boolean,
+    onByDefault: Boolean,
 ) :
     AutoconsentSettingsDataStore {
 
     private val preferences: SharedPreferences by lazy { context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE) }
+    private var internalUserSettings: Boolean
+
+    init {
```

Comment:
I don't follow the logic to add `internalUserSettings` here. If the idea is knowing if the user had enabled autoconsent then you can use isSettingEnabled() from the public API meaning the data store doesn't need any changes.

Reply:
This is related to this [comment](https://github.com/duckduckgo/Android/pull/4111#discussion_r1465032027) Aitor left on a previous PR. Since we're trying to minimize the impact on performance, we want to avoid reading sharedpreferences every time we need to fire a page loaded pixel. Therefore, I'm using that variable to cache the value from sharedprefs